org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractContentAssistTest.java

```diff
@@ -72,9 +83,18 @@ public XtextResource getResourceFor(InputStream stream) {
 		XtextResourceSet resourceSet = resourceSetProvider.get();
 		initializeTypeProvider(resourceSet);
 		try {
-			URI resourceUri = URI.createURI("Test." + fileExtensionProvider.getPrimaryFileExtension());
-			Resource resource = resourceSet.createResource(resourceUri);
-			resource.load(stream, null);
+			String projectFullPath = javaProject.getProject().getFullPath().toString();
+			URI resourceUri = URI.createPlatformResourceURI(projectFullPath + "/" + "Test." + fileExtensionProvider.getPrimaryFileExtension(), true);
```

Comment:
am not sure. dont in java projects. model files need to reside in source folder?

Reply:
Exactly! That's why I was suggesting below to allow for relative path specification.

Reply:
@miklossy I still don't understand this part... see my comment and @cdietrich's one

Reply:
from my pov this is still open

Reply:
@cdietrich relative paths are allowed, see the other answers. What did you mean?

Reply:
i mean the default. unfortunately i dont have cappa to try it

Reply:
Ok, now maybe I see: the file on which the content assistant is triggered! That's always the same and cannot be created in a specific folder. Is that right @miklossy ? If it's so, then we should allow for some customization? Maybe a variable that can be changed? And maybe the default should be "src/Test"?

Reply:
@miklossy Thanks!
@cdietrich I think this also fixes your last issue.